akka-diagnostics/src/main/scala/akka/diagnostics/ConfigChecker.scala

```diff
@@ -0,0 +1,1169 @@
+            "a thread-pool-executor instead.")
+        else Nil
+      }
+    }
+
+  private def checkNumberOfDispatchers(dispatchers: Map[String, Config]): List[ConfigWarning] =
+    ifEnabled("dispatcher-count") { checkerKey =>
+      val customDispatchers = dispatchers.collect { case (p, _) if !isLightbendInternalDispatcher(p) => p }
+      if (customDispatchers.size > 6)
+        warn(
+          checkerKey,
+          customDispatchers.toList,
+          s"You have configured [${customDispatchers.size}] different custom dispatchers. " +
+          "Do you really need that many dispatchers? " +
+          "Separating into CPU bound tasks and blocking (IO) tasks are often enough.")
+      else Nil
+    }
+
+  private def isLightbendInternalDispatcher(configPath: String): Boolean =
+    knownDispatcherPrefixes.exists(prefix => configPath.startsWith(prefix))
+
+  private def checkTotalDispatcherPoolSize(dispatchers: Map[String, Config]): List[ConfigWarning] =
+    ifEnabled("dispatcher-total-size") { checkerKey =>
+      val sizes = dispatchers.collect {
+        // FIXME does the filtering really make sense here, don't we want to include the lightbend
+        // dispatcher pool sizes as well?
+        case (p, c) if !isLightbendInternalDispatcher(p) =>
+          val cfgWithFallback = c.withFallback(system.dispatchers.defaultDispatcherConfig)
+          p -> Try(dispatcherPoolSize(cfgWithFallback)).getOrElse(0)
+      }
+      val total = sizes.foldLeft(0) { case (acc, (_, s)) => acc + s }
+      val availableProcessors = Runtime.getRuntime.availableProcessors
+      if (total > 200 && total > availableProcessors * 2)
+        warn(
+          checkerKey,
+          sizes.collect { case (p, s) if s != 0 => p }.toList,
+          s"You have a total of [$total] threads in all configured dispatchers. " +
+          "That many threads might result in reduced performance. " +
+          s"This machine has [$availableProcessors] available processors.")
+      else Nil
+    }
+
+  private def isRemoteConfigAvailable: Boolean = {
+    val provider = config.getString("akka.actor.provider")
+    // check existence of a property from reference.conf that will unlikely be defined elsewhere
+    reference.hasPath("akka.actor.serializers.daemon-create") &&
+    (provider == "remote" || provider == "akka.remote.RemoteActorRefProvider" || isClusterConfigAvailable)
+  }
+
+  private def remoteConfigPath(path: String): String = {
+//    if (isAkka26)
+//      "akka.remote.classic" + path.substring(11)
+//    else
+    path
+  }
+
+  private def checkRemote(): Vector[ConfigWarning] =
+    if (isRemoteConfigAvailable) {
+      Vector.empty[ConfigWarning] ++
+      checkRemoteDispatcher() ++
+      checkSecureCookie() ++
+//      checkTransportFailureDetector() ++
+      checkRemoteWatchFailureDetector() ++
+//      checkRetryGate() ++
+//      checkPruneQuarantine() ++
+//      checkEnabledTransports() ++
```

Comment:
yes, all these that are commented out are for classic remoting. We can remove checks for classic remoting.

Reply:
removed